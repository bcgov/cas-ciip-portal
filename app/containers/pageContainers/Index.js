import React, {Component} from 'react';
import {Button, Container, Row, Col, Card, Jumbotron} from 'react-bootstrap';
import Link from 'next/link';
import DefaultLayout from '../../layouts/default-layout';

const heading = {
  color: '#003366',
  fontSize: 'Bold 25px',
  fontWeight: 'bold',
  textAlign: 'center'
};

const Cards = {
  marginTop: '50px',
  marginBottom: '25px',
  borderRadius: '15px',
  textAlign: 'center'
};

const CardBody1 = {
  backgroundColor: '#003366',
  padding: '50px',
  borderRadius: '15px'
};
const CardTitle1 = {
  color: 'white',
  fontSize: '25px',
  marginBottom: '15px',
  fontWeight: 'bold'
};
const CardTitle2 = {
  fontSize: '25px',
  color: '#003366',
  marginBottom: '15px',
  fontWeight: 'bold'
};
export default class Index extends Component {
  render() {
    return (
      <DefaultLayout>
        <Jumbotron>
          <h1 style={heading}>Welcome to the CIIP Portal</h1>
        </Jumbotron>
        <Container>
          <Row>
            <Col md>
              <Card style={Cards}>
                <Card.Body style={CardBody1}>
                  <Card.Title style={CardTitle1}>Login</Card.Title>
                  <Card.Text style={{color: 'white'}}>
                    If you have already registered, click below to log in.
                  </Card.Text>
                  <Link
                    href={{
                      pathname: '/user-dashboard',
                      query: {userId: 1, id: 'WyJ1c2VycyIsMV0='}
                    }}
                  >
                    <Button
                      style={{backgroundColor: '#EDA500', color: '#003366'}}
                      size="lg"
                    >
                      Login
                    </Button>
                  </Link>
                </Card.Body>
              </Card>
            </Col>
            <Col>
              <Card style={Cards}>
                <Card.Body style={{padding: '50px'}}>
                  <Card.Title style={CardTitle2}>Register for CIIP</Card.Title>
                  <Card.Text style={{color: '#003366'}}>
                    Click here to create your industrial reporter account.
                  </Card.Text>
                  <Link href="/registration">
                    <Button
                      style={{backgroundColor: '#003366'}}
                      size="lg"
                      onClick={this.showModal}
                    >
                      Register for CIIP
                    </Button>
                  </Link>
                </Card.Body>
              </Card>
            </Col>
          </Row>
        </Container>
      </DefaultLayout>
    );
  }
}
