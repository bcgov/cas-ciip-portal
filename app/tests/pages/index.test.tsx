import React from 'react';
import {shallow} from 'enzyme';
import Index from '../../pages';

describe('landing', () => {
  it('It matches the last accepted Snapshot', () => {
    const wrapper = shallow(<Index />);
    expect(wrapper).toMatchSnapshot();
  });
});
