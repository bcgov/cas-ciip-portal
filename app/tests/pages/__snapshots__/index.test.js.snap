// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing It matches the last accepted Snapshot 1`] = `
<DefaultLayout>
  <Jumbotron
    fluid={false}
  >
    <h1
      style={
        Object {
          "color": "#003366",
          "fontSize": "Bold 25px",
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Welcome to the CIIP Portal
    </h1>
  </Jumbotron>
  <Container
    fluid={false}
  >
    <ForwardRef
      noGutters={false}
    >
      <Col
        md={true}
      >
        <Card
          body={false}
          style={
            Object {
              "borderRadius": "15px",
              "marginBottom": "25px",
              "marginTop": "50px",
              "textAlign": "center",
            }
          }
        >
          <CardBody
            style={
              Object {
                "backgroundColor": "#003366",
                "borderRadius": "15px",
                "padding": "50px",
              }
            }
          >
            <CardTitle
              style={
                Object {
                  "color": "white",
                  "fontSize": "25px",
                  "fontWeight": "bold",
                  "marginBottom": "15px",
                }
              }
            >
              Login
            </CardTitle>
            <CardText
              style={
                Object {
                  "color": "white",
                }
              }
            >
              If you have already registered, click below to log in.
            </CardText>
            <Link
              href={
                Object {
                  "pathname": "/user-dashboard",
                  "query": Object {
                    "id": "WyJ1c2VycyIsMV0=",
                    "userId": 1,
                  },
                }
              }
            >
              <Button
                active={false}
                disabled={false}
                size="lg"
                style={
                  Object {
                    "backgroundColor": "#EDA500",
                    "color": "#003366",
                  }
                }
                type="button"
                variant="primary"
              >
                Login
              </Button>
            </Link>
          </CardBody>
        </Card>
      </Col>
      <Col>
        <Card
          body={false}
          style={
            Object {
              "borderRadius": "15px",
              "marginBottom": "25px",
              "marginTop": "50px",
              "textAlign": "center",
            }
          }
        >
          <CardBody
            style={
              Object {
                "padding": "50px",
              }
            }
          >
            <CardTitle
              style={
                Object {
                  "color": "#003366",
                  "fontSize": "25px",
                  "fontWeight": "bold",
                  "marginBottom": "15px",
                }
              }
            >
              Register for CIIP
            </CardTitle>
            <CardText
              style={
                Object {
                  "color": "#003366",
                }
              }
            >
              Click here to create your industrial reporter account.
            </CardText>
            <Link
              href="/registration"
            >
              <Button
                active={false}
                disabled={false}
                size="lg"
                style={
                  Object {
                    "backgroundColor": "#003366",
                  }
                }
                type="button"
                variant="primary"
              >
                Register for CIIP
              </Button>
            </Link>
          </CardBody>
        </Card>
      </Col>
    </ForwardRef>
  </Container>
</DefaultLayout>
`;
